src/nnet3/nnet-simple-component.h

```diff
@@ -309,6 +310,7 @@ class AffineComponent: public UpdatableComponent {
   Component *CollapseWithNext(const FixedAffineComponent &next) const;
```

Comment:
About the second part, I'm not sure how to check Scale(), SetZero(), and Add(). This is the code I have come up with:
void TestNnetComponentVectorizeUnVectorize(Component *c) {
  UpdatableComponent *uc = dynamic_cast<UpdatableComponent*>(c);
  if((uc==NULL) || (uc->NumParameters() == 0))
    return;
  Vector<BaseFloat> params;
  UpdatableComponent *uc2 = dynamic_cast<UpdatableComponent*>(uc->Copy());
  uc2->SetZero(false);
  uc2->Vectorize(&params);
  KALDI_ASSERT(params.Min()==0.0 && params.Sum()==0.0);
  uc->Vectorize(&params);
  uc2->UnVectorize(params);
  KALDI_ASSERT(uc2->Info() == uc->Info());
  KALDI_ASSERT(uc2->DotProduct(*uc2) == uc->DotProduct(*uc));
  delete uc2;
}
void TestNnetComponentUpdatableFlag(Component *c) {
  UpdatableComponent *uc = dynamic_cast<UpdatableComponent*>(c);
  if(uc==NULL)
    return;
  if(!(uc->Properties() & kUpdatableComponent)){
    KALDI_ASSERT(uc->NumParameters() == 0);
    KALDI_ASSERT(uc->DotProduct(*uc) == 0);
    UpdatableComponent *uc2 = dynamic_cast<UpdatableComponent*>(uc->Copy());
    uc2->Scale(7.0);
    uc2->Add(3.0, *uc);
    KALDI_ASSERT(uc2->Info() == uc->Info());
    uc->SetZero(false);
    KALDI_ASSERT(uc2->Info() == uc->Info());
    delete uc2;
  }
}

Also this test caused Convolutional1dComponent to fail in Vectorize test because it does not implement it. Should I fix that too?